lib/core/src/Cardano/Wallet/DB/Sqlite/Types.hs

```diff
@@ -393,12 +393,12 @@ instance PersistField AddressPoolGap where
     toPersistValue = toPersistValue . getAddressPoolGap
     fromPersistValue pv = fromPersistValue >=> mkAddressPoolGap' $ pv
       where
-        mkAddressPoolGap' :: Word8 -> Either Text AddressPoolGap
+        mkAddressPoolGap' :: Word32 -> Either Text AddressPoolGap
         mkAddressPoolGap' = first msg . mkAddressPoolGap . fromIntegral
         msg e = T.pack $ "not a valid value: " <> show pv <> ": " <> show e
 
 instance PersistFieldSql AddressPoolGap where
-    sqlType _ = sqlType (Proxy @Word8)
+    sqlType _ = sqlType (Proxy @Word32)
```

Comment:
So this corresponds to the newtype

```
newtype AddressPoolGap = AddressPoolGap
    { getAddressPoolGap :: Word32 }
    deriving stock (Generic, Show, Eq, Ord)
```

I'm not sure if there is a way to get the inner type of a newtype. Maybe with generics?

Reply:
```hs
{-# LANGUAGE DeriveGeneric #-}
{-# LANGUAGE FlexibleContexts #-}
{-# LANGUAGE GADTs #-}
{-# LANGUAGE KindSignatures #-}
{-# LANGUAGE ScopedTypeVariables #-}
{-# LANGUAGE TypeApplications #-}
{-# LANGUAGE TypeFamilies #-}

import Data.Proxy
import Data.Typeable
import Data.Word
import GHC.Generics

newtype AddressPoolGap = AddressPoolGap
    { getAddressPoolGap :: Word32 }
    deriving (Generic, Show)

innerTypeOf :: (Generic a, GInnerType (Rep a)) => a -> Proxy (Inner (Rep a))
innerTypeOf = gInnerType . from

class GInnerType (f :: * -> *) where
    type Inner f :: *
    gInnerType :: f a -> Proxy (Inner f)

instance (GInnerType f) => GInnerType (M1 i c f) where
    type Inner (M1 i c f) = Inner f
    gInnerType (M1 fp) = gInnerType fp

instance GInnerType (K1 i c) where
    type Inner (K1 i c) = c
    gInnerType (K1 c) = mkProxy c
      where
        mkProxy :: forall a. a -> Proxy a
        mkProxy _ = Proxy @a

main :: IO ()
main = print $ typeRep $ innerTypeOf $ AddressPoolGap 42
```

```
$ stack runhaskell Main.hs
Word32
```

Reply:
Want it only on type level.

I think this will do:
```
type family Inner a where
  Inner (M1 _ _ x) = Inner x
  Inner (K1 _ x) = x
```